[FIX] git_trunk: ignore submodule changes

Drsmartinkus opened the pull request:
Fix issue when git stash (git trunk refresh) was causing 'No stash entries found.' error
due to the fact that submodule contains changes

[BRANCH] bugfix/stash-apply-dma

oerp-odoo commented on git_trunk/git_trunk.py:
```diff
@@ -1034,7 +1034,7 @@ def run(self, **kwargs):
         """
         super().run(**kwargs)
-        if self.git.diff('--stat'):
+        if self.git.diff('--ignore-submodules'):
```
Should probably leave comment explaining why need to use this flag.

oerp-odoo commented on git_trunk/tests/common.py:
```diff
@@ -44,6 +44,23 @@ def _build_repo(self, git, dir_local, dir_remote):
         git.checkout('master')
 
+    def _create_dummy_submodule(self, dir_name):
+        """
```
Looks like you are not using git-hooks:)
This is incorrect docstring format.

It violates pydocstyle: `D200 One-line docstring should fit on one line with quotes`

oerp-odoo commented on git_trunk/tests/test_git_trunk_refresh.py:
```diff
@@ -78,3 +78,15 @@ def test_02_git_refresh(self):
+            repo_path=self.dir_local.name,
+            log_level=common.LOG_LEVEL,
+        )
+        trunk_refresh.run()
```
But where is actual test? You run refresh, but you do not check if refresh was done correctly. It could be the fact that refresh fails silently and does not do refresh correctly.

If you would check other refresh tests, you would see there is helper `_test_git_refresh`. You could use it too, to make sure refresh does what its supposed to do.

Drsmartinkus replied:
I added a diff check before and after refresh, because once submodule is added diff is dirty and its keeps dirty after refresh is completed, you think its good or still would need to improve it?

oerp-odoo replied:
It doesnt matter its dirty or not. Diff should still be the same before refresh and after (cause you use plain diff check in both cases).

oerp-odoo commented:
Also you forgot to bump version

Drsmartinkus committed:
[FIX] git_trunk: ignore submodule changes

Fix issue when git stash (git trunk refresh) was causing 'No stash entries found.' error
due to the fact that submodule contain changes

[BRANCH] bugfix/stash-apply-dma

oerp-odoo commented on git_trunk/tests/test_git_trunk_refresh.py:
```diff
@@ -78,3 +78,15 @@ def test_02_git_refresh(self):
+            log_level=common.LOG_LEVEL,
+        )
+        trunk_refresh.run()
+        diff = self.git.diff()
```
diff must be saved before you run refresh, so you would know its the same as was before doing it. Now what you are doing is comparing same value with same value (means it will never be different) or in other words false positive.

oerp-odoo commented on git_trunk/__init__.py:
```diff
@@ -1 +1 @@
-__version__ = '0.2.0'
+__version__ = '0.3.0'
```
its a fix, bump patch version.
